engine/src/main/java/com/agiletec/aps/system/services/i18n/I18nManagerWrapper.java

```diff
@@ -43,9 +59,15 @@ public I18nManagerWrapper(String langCode, II18nManager i18nManager) {
 	 */
 	public String getLabel(String key) throws EntException {
 		String label = null;
-		if (key != null) {
-			label = this._i18nManager.getLabel(key, this._lang);
+		if (null != key) {
+			label = this.i18nManager.getLabel(key, this.currentLangCode);
+            if (StringUtils.isBlank(label) && null != reqCtx) {
```

Comment:
this `if` is obscure because you control reqCtx but don't use it (btw we should discourage the use of global variables), at least pls move the check for null inside the method.

Reply:
Fixed